src/lib/repository.ts

```diff
@@ -293,5 +297,6 @@ export const createRepository = <E extends Entity, M extends Metadata<E>>(model:
         update: (filter: Filters<E, M>, data?: Partial<E>, options: RepositoryMethodOptions = {}) => update(model, filter, data, options),
         delete: (filter?: Filters<E, M>, options?: RepositoryMethodOptions) => remove(model, filter, options),
         createBulk: (dataItems: Array<Partial<E>>, options?: RepositoryMethodOptions) => createBulk(model, dataItems, options),
+        _getQueryBuilder: (filter?: Filters<E, M>, options?: RepositoryListOptions<E, M>) => _getQueryBuilder(model, filter, options),
```

Comment:
Generally I think it the API would be better if we used an option and handle it in serializers:
 1. You could use it for all methods (create, detail, ...)
 2. You would not need to add a new method for each extension.
 
However this would be terribly difficult to type (similiar issue we had with `count: true` option, yielding different results).

I would suggest at very least to name it after list. It is not a generic `getQueryBuilder` for all methods, but only list (possibly detail). It might avoid confusion with devs trying to find a way to pass the update data to the method for example.

`_getListQueryBuilder`?

I am not especially fond of the `_` prefix. I don't mind it, but I don't think there's a reason for it, it is part of public API that is meant to be used by the devs and the QB distinction is already in the name.

Reply:
Thanks for the input! 🙇 

I am fine with `getListQueryBuilder`, I don't have a strong preference for the `_` prefix - I'll rename.

As for the suggestion to have it as an `option` - I think that would be nice, could possibly be as a later refactor or similar? Since it sounds like a larger change. Am open to suggestions, though.

Reply:
+1 getListQueryBuilder

Reply:
thanks for the update